components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/SimulatorAPITestcase.java

```diff
@@ -0,0 +1,316 @@
+import org.ops4j.pax.exam.Option;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+import org.ops4j.pax.exam.testng.listener.PaxExam;
+import org.osgi.framework.BundleContext;
+import org.testng.Assert;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+import org.wso2.carbon.analytics.test.osgi.util.HTTPResponseMessage;
+import org.wso2.carbon.analytics.test.osgi.util.TestUtil;
+import org.wso2.carbon.container.CarbonContainerFactory;
+import org.wso2.carbon.kernel.CarbonServerInfo;
+
+import java.io.File;
+import java.net.URI;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import javax.inject.Inject;
+
+import static org.wso2.carbon.container.options.CarbonDistributionOption.carbonDistribution;
+import static org.wso2.carbon.container.options.CarbonDistributionOption.copyFile;
+
+@Listeners(PaxExam.class)
+@ExamReactorStrategy(PerClass.class)
+@ExamFactory(CarbonContainerFactory.class)
+public class SimulatorAPITestcase {
+    private static final org.apache.log4j.Logger logger = org.apache.log4j.Logger.getLogger(SimulatorAPITestcase.class);
+
+    private static final int HTTP_PORT = 9390;
+    private static final String HOSTNAME = "localhost";
+    private URI baseURI = URI.create(String.format("http://%s:%d", HOSTNAME, HTTP_PORT));
+
+    private final String DEFAULT_USER_NAME = "admin";
+    private final String DEFAULT_PASSWORD = "admin";
+
+    private static final String validSingleEventConfig = "{\n" +
+            "  \"streamName\": \"CountStream\",\n" +
+            "  \"siddhiAppName\": \"ReceiveAndCount\",\n" +
+            "  \"data\": ['Customer1', 26, 'USA', 15 ]\n" +
+            "}";
+    private static final String inValidSingleEventConfig = "{\n" +
+            "  \"streamName\": \"CountStream\",\n" +
+            "  \"data\": ['Customer1', 26, 'USA', 15 ]\n" +
+            "}";
+
+    private static final String validFeedEventConfig = "{\"properties\":{\"simulationName\":\"FeedSimulation\"," +
+            "\"startTimestamp\":\"\",\"endTimestamp\":\"\",\"noOfEvents\":\"\",\"description\":\"\"," +
+            "\"timeInterval\":\"1000\"},\"sources\":[{\"siddhiAppName\":\"ReceiveAndCount\",\"streamName\":" +
+            "\"CountStream\",\"timestampInterval\":\"1000\",\"simulationType\":\"CSV_SIMULATION\",\"fileName\":" +
+            "\"sampleCSV.csv\",\"delimiter\":\",\",\"isOrdered\":true,\"indices\":\"0,1,2,3\"}]}";
+
+    @Inject
+    private CarbonServerInfo carbonServerInfo;
+
+    @Inject
+    protected BundleContext bundleContext;
+
+    @Configuration
+    public Option[] createConfiguration() {
+        return new Option[]{
+                carbonDistribution(Paths.get("target", "wso2das-" +
+                        System.getProperty("carbon.analytic.version")), "editor"),
+                copySiddhiFileOption(),
+                copyCSVFileOption(),
+                //CarbonDistributionOption.debug(5005)
```

Comment:
Shall we remove the unnecessary comments?

Reply:
Fixed with commit [#f9917365bd8f8a1c65a09dace8e464fae8933ec3](https://github.com/wso2/carbon-analytics/pull/1259/commits/f9917365bd8f8a1c65a09dace8e464fae8933ec3)